src/System.Runtime.Serialization.Formatters/tests/EqualityExtensions.cs

```diff
@@ -130,5 +135,764 @@ public static bool IsEqual(this StringDictionary @this, StringDictionary other)
+        {
+            if (!(@this != null &&
+                other != null &&
+                @this.Capacity == other.Capacity &&
+                @this.Count == other.Count &&
+                @this.IsFixedSize == other.IsFixedSize &&
+                @this.IsReadOnly == other.IsReadOnly &&
+                @this.IsSynchronized == other.IsSynchronized))
+                return false;
+
+            for (int i = 0; i < @this.Count; i++)
+            {
+                if (!BinaryFormatterTests.CheckEquals(@this[i], other[i]))
+                    return false;
+            }
+
+            return true;
+        }
+
+        public static bool IsEqual(this BitArray @this, BitArray other)
+        {
+            if (!(@this != null &&
+                other != null &&
+                @this.Length == other.Length &&
+                @this.Count == other.Count &&
+                @this.IsReadOnly == other.IsReadOnly &&
+                @this.IsSynchronized == other.IsSynchronized))
+                return false;
+
+            return BinaryFormatterTests.CheckSequenceEquals(@this, other);
+        }
+
+        public static bool IsEqual(this Dictionary<int, string> @this, Dictionary<int, string> other)
+        {
+            if (!(@this != null &&
+                other != null &&
+                BinaryFormatterTests.CheckEquals(@this.Comparer, other.Comparer) &&
+                @this.Count == other.Count &&
+                BinaryFormatterTests.CheckSequenceEquals(@this.Keys, other.Keys) &&
+                BinaryFormatterTests.CheckSequenceEquals(@this.Values, other.Values)))
+                return false;
+
+            foreach (var kv in @this)
+            {
+                if (@this[kv.Key] != other[kv.Key])
+                    return false;
+            }
+
+            return true;
+        }
+
+        public static bool IsEqual(this PointEqualityComparer @this, PointEqualityComparer other)
+        {
+            return @this != null &&
+                other != null;
+        }
+
+        public static bool IsEqual(this HashSet<Point> @this, HashSet<Point> other)
+        {
+            if (!(@this != null &&
+                other != null &&
+                @this.Count == other.Count &&
+                BinaryFormatterTests.CheckEquals(@this.Comparer, other.Comparer)))
+                return false;
+
+            return BinaryFormatterTests.CheckSequenceEquals(@this, other);
+        }
+
+        public static bool IsEqual(this LinkedListNode<Point> @this, LinkedListNode<Point> other)
+        {
+            if (@this == null && other == null)
+                return true;
+
+            return @this != null
+                && other != null &&
+                BinaryFormatterTests.CheckEquals(@this.Value, other.Value);
+        }
+
+        public static bool IsEqual(this LinkedList<Point> @this, LinkedList<Point> other)
+        {
+            if (!(@this != null &&
```

Comment:
I'm only fixing when I understand why the value is null, in many cases it should not be null even though that would be correct to check it in here